min-max/min-max-sum/chandan-min-max-sum.js

```diff
@@ -0,0 +1,57 @@
+class MaxMin {
+    constructor() {
+        console.log("Constructor initialized!");
+        this.arr = arr;
+    }
+    getMin() {
+        var min = this.arr[0];
+        for(var i=1; i<this.arr.length; i++) {  
+            if(min > this.arr[i]) {
+                min = this.arr[i];
+            }
+        }
+        return min;
+    }
+    getMax() {
+        var max = this.arr[0];
+        for(var i=1; i<this.arr.length; i++) {  
+            if(max < this.arr[i]) {
+                max = this.arr[i];
+            }
+        }
+        return max;
+    }
+}
+
+class Sum extends MaxMin  {
+    getSum() {
+    var sum = 0;
+    for(var i=0; i<this.arr.length; i++) {
+        sum += this.arr[i];
+     }
+     return sum;
+    }   
+}
+
+function getMaxOrMin(arr, option) {
+    var num = new Sum(arr);
```

Comment:
can you explain how `arr` parameter is assigning to the class `this.arr` variable?

Reply:
as sum class extended the MaxMin class it's automatically getting the array or should I set the constructor using super keyword?